apps/src/code-studio/videos.js

```diff
@@ -226,36 +225,29 @@ videos.showVideoDialog = function(options, forceShowVideo) {
     'text-align': 'right'
   });
   nav.append(fallbackPlayerLinkDiv);
+  function onResize() {
+    const containerDimensions = getVideoContainerDimensions();
+    $div.height(containerDimensions.containerHeight);
+    $div.width(containerDimensions.containerWidth);
+
+    // Standard css hack to center a div within the viewport.
+    $div.css({
+      top: '50%',
+      left: '50%',
+      marginTop: containerDimensions.containerHeight / -2 + 'px',
+      marginLeft: containerDimensions.containerWidth / -2 + 'px'
+    });
 
-  // Resize modal to fit constraining dimension.
-  let accurateInnerHeight = Math.min(window.innerHeight, window.outerHeight);
-  let accurateInnerWidth = Math.min(window.innerWidth, window.outerWidth);
-  let windowHeight = Math.min(accurateInnerHeight, accurateInnerWidth);
-  let windowWidth = Math.max(accurateInnerHeight, accurateInnerWidth);
-
-  var height = windowHeight * widthRatio,
-    width = windowWidth * heightRatio;
-
-  if (height * aspectRatio < width) {
-    $div.height(height);
-    $div.width(height * aspectRatio);
-  } else {
-    $div.height(width / aspectRatio);
-    $div.width(width);
+    // The nav bar will have changed dimensions due to the resizing of the
+    // container. We re-calculate the available height for the video to account
+    // for this.
+    const availableHeight = getVideoHeight();
+    $(video).height(availableHeight);
+    notesDiv.height(availableHeight);
   }
 
-  // Standard css hack to center a div within the viewport.
-  $div.css({
-    top: '50%',
-    left: '50%',
-    marginTop: $div.height() / -2 + 'px',
-    marginLeft: $div.width() / -2 + 'px'
-  });
-
-  var divHeight = $div.innerHeight() - nav.outerHeight();
-  $(video).height(divHeight);
-
-  notesDiv.height(divHeight);
+  window.addEventListener('resize', onResize);
```

Comment:
We normally throttle this kind of callback.

Reply:
@breville Do you mean listeners to 'resize' or listeners in general? (Or something else?)

Reply:
I've seen us throttle a lot of resize listeners in particular, such as here:

https://github.com/code-dot-org/code-dot-org/blob/33c27d7b5a59d422b37fbedd6a2d2b9dd6afc085/apps/src/templates/RotateContainer.jsx#L57-L58